File: C++/0009-Palindrome-Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string test = to_string(x);
        int startIndex = 0;
        int endIndex = test.length() - 1;
        while (test[startIndex] == test[endIndex]) {
            if (startIndex < test.length() - 1) {
                startIndex++;
            }
            else {
                break;
            }
            if (endIndex >= 0) {
                endIndex--;
            }
            else {
                break;
            }
        }
        if (startIndex == test.length() - 1) {
            return true;
        }
        return false;
    }
};

File: Java/0009-Palindrome-Number.java
class Solution {
    public boolean isPalindrome(int x) {
        String xString = String.valueOf(x);
        int left = 0;
        int right = xString.length() - 1;
        while (left < right) {
            if (xString.charAt(left) != xString.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}